test/__test__/utils/drbUtils.test.tsx

```diff
@@ -0,0 +1,109 @@
+      expect(
+        checkQueryParamsEquality(testQuery, { q: "toast", field: "title" })
+      ).toBe(true)
+    })
+
+    it("should handle keyword & subject query", () => {
+      const testQuery = `?query=keyword%3Atoast,subject%3ASnacks&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}`
+      expect(
+        checkQueryParamsEquality(testQuery, {
+          q: "toast",
+          filters: { subjectLiteral: "Snacks" },
+        })
+      ).toBe(true)
+    })
+
+    it("should handle contributor filter", () => {
+      const testQuery = `?query=keyword%3A*,author%3APoe&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}`
+      expect(
+        checkQueryParamsEquality(testQuery, {
+          filters: { contributorLiteral: "Poe" },
+        })
+      ).toBe(true)
+    })
+
+    it("should handle lang filter", () => {
+      const testQuery = `?query=keyword%3A*&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}&filter=language%3Aen`
+      expect(
+        checkQueryParamsEquality(testQuery, {
+          filters: { language: "lang:en" },
+        })
+      ).toBe(true)
+    })
+
+    it("should handle dateAfter filter", () => {
+      const testQuery = `?query=keyword%3A*&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}&filter=startYear%3A2000`
+      expect(
+        checkQueryParamsEquality(testQuery, { filters: { dateAfter: "2000" } })
+      ).toBe(true)
+    })
+
+    it("should handle dateBefore filter", () => {
+      const testQuery = `?query=keyword%3A*&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}&filter=endYear%3A2020`
+      expect(
+        checkQueryParamsEquality(testQuery, { filters: { dateBefore: "2020" } })
+      ).toBe(true)
+    })
+
+    it("should handle dateAfter & dateBefore filter", () => {
+      const testQuery = `?query=keyword%3A*&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}&filter=startYear%3A2000,endYear%3A2020`
+      expect(
+        checkQueryParamsEquality(testQuery, {
+          filters: { dateAfter: "2000", dateBefore: "2020" },
+        })
+      ).toBe(true)
+    })
+
+    it("should handle contributor param", () => {
+      const testQuery = `?query=keyword%3A*%2Cauthor%3APoe&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}`
+      expect(checkQueryParamsEquality(testQuery, { contributor: "Poe" })).toBe(
+        true
+      )
+    })
+
+    it("should handle title param", () => {
+      const testQuery = `?query=keyword%3A*%2Ctitle%3ARaven&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}`
+      expect(checkQueryParamsEquality(testQuery, { title: "Raven" })).toBe(true)
+    })
+
+    it("should handle subject param", () => {
+      const testQuery = `?query=keyword%3A*%2Csubject%3Acorvids&page=1&source=catalog&size=${DRB_RESULTS_PER_PAGE}`
+      expect(checkQueryParamsEquality(testQuery, { subject: "corvids" })).toBe(
+        true
+      )
+    })
+
+    it("should combine query params", () => {
```

Comment:
The more I look at this file the more complex the query object vs the url query is. All seems fine but just pointing out the complexity.

Reply:
how so?